01projects/sec04CMakeHelloWorld.md

```diff
@@ -0,0 +1,237 @@
+
+``` cmake
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+```
+
+#### Parsing nested `CMakeLists.txt` within the `src/` directory
+
+Until now, we have set certain high-level description of the project, but have not provided instructions on building the "hello_world" executable program from the source!
+This is intentional.
+The task of describing the build of each logical/modular project unit is delegated to nested/leaf `CMakeLists.txt` that reflects the project's hierarchy.
+In this case, we wish to 'hand-off' control to `src/CMakeLists.txt` for building the `hello_world` executable.
+This is achieved by using the [`add_subdirectory()`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command.
+
+``` cmake
+add_subdirectory(src)
+```
+
+In the above line, we instruct CMake to process the `CMakeLists.txt` located within the `src/` directory.
+These list files can be nested quite deeply, and at each depth level, we can have as many 'sibling' folders with one `CMakeLists.txt` per folder that describes the logical program unit to be built.
+
+After processing all nested list files, control returns to the top level `CMakeLists.txt`.
+Since the above line was the last in this file, the configuration process is completed.
+We now focus on the build details that is to be described in `src/CMakeLists.txt`.
+
+### Contents of `src/CMakeLists.txt`
+
+Within this 'leaf' `CMakeLists.txt` file, we describe how the `hello_world` executable is to be built.
+This is considered best practice, i.e. to have a `CMakeLists.txt` as close to the relevant source files for each modular functionality within the project.
+This file can see and use all the properties, variables, functions and macros defined at any `CMakeLists.txt` in its parent scope (in this case, all definitions from the top-level `CMakeLists.txt` file are visible here i.e. they are in the 'scope' of this leaf file).
+
+#### Targets and usage requirements
+
+We like CMake to build for us an executable called 'hello_world'.
+In this case, `hello_world` is a CMake 'target' (or a CMake 'executable target' to be precise).
+Targets (not variables) are the first-class citizens in modern CMake language.
+Hence, the relationships between various constituents (i.e. executables, source files, header files, internal libraries, other third-party code/files, system dependencies etc.) is to be described in a target-centric way within each `CMakeLists.txt` file.
+
+The behaviour of targets, their features and their various properties can be set in three ways:
+- only on/for/affecting itself (using the PRIVATE keyword)
+- only on/for/affecting other targets that consume this current target (using the INTERFACE keyword)
+- on/for/affecting itself as well as other targets that consume this current target (using the INTERFACE keyword)
```

Comment:
I think the last bullet point should be "PUBLIC", not interface

Reply:
Fixed